### PR TITLE
lorri: add nixPackage option; add notifier

### PR DESCRIPTION
**Pull request by nyarly**
### Description

Adds `nixPackage` option, which allows lorri to use `nixUnstable`,
and therefore flake based shell.s

Adds a little notifier service, so that starts and ends of builds can be reported.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`. _bash.nix test fails_

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/services/lorri.nix [hunk 1]
```diff
@@ -60,6 +90,23 @@ in {
 
         Install = { WantedBy = [ "sockets.target" ]; };
       };
+
+      services.lorri-notify = mkIf (cfg.notify) {
```
Comment (berbiche) [inline modules/services/lorri.nix:94]:
```suggestion
      systemd.user.services.lorri-notify = mkIf (pkgs.stdenv.hostPlatform.isLinux && cfg.notify) {
```

Until we have a common backend to declare services for Darwin and Linux, the service will have to be gated.

File: modules/services/lorri.nix [hunk 2]
```diff
@@ -6,18 +6,48 @@ let
...
       description = "Which lorri package to install.";
     };
+
+    nixPackage = mkOption {
+      type = types.package;
+      default = pkgs.nix;
+      defaultText = literalExpression "pkgs.nix";
+      description = "Which nix package to use.";
+      example = literalExpression "pkgs.nixUnstable";
+    };
```
Comment (berbiche) [inline modules/services/lorri.nix:50]: TODO: change the default value to `config.nix.package` once we add this option.

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks!

I didn't test the notify script but I found your comment on the lorri issue tracker (https://github.com/nix-community/lorri/issues/77#issuecomment-1035564228) so I assume it works correctly!

File: modules/services/lorri.nix (hunk 2 above)
Comment (nyarly) [reply modules/services/lorri.nix:50]: Not sure how to interpret this - just a note to the project, or does it need to be addressed?

File: modules/services/lorri.nix [hunk 3]
```diff
@@ -60,6 +90,30 @@ in {
 
         Install = { WantedBy = [ "sockets.target" ]; };
       };
+
+      services.lorri-notify =
+        mkIf (pkgs.stdenv.hostPlatform.isLinux && cfg.notify) {
```
Comment (rycee) [inline modules/services/lorri.nix:95]: No need to check for Linux platform
```suggestion
        mkIf cfg.notify {
```

File: modules/services/lorri.nix [hunk 4]
```diff
@@ -6,18 +6,48 @@ let
...
+    name = "lorri_notify";
+    executable = true;
+    text = ''
+      #! /usr/bin/env bash
+
+      lorri internal stream-events --kind live |\
+      jq --unbuffered '${jqFile}' |\
+      xargs -n 1 notify-send "Lorri Build"
+    '';
+  };
```
Comment (rycee) [inline modules/services/lorri.nix:27]: Can simplify using `writeShellScript`:
```suggestion
  notifyScript = pkgs.writeShellScript "lorri-notify" ''
    lorri internal stream-events --kind live \
      | jq --unbuffered '${jqFile}' \
      | xargs -n 1 notify-send "Lorri Build"
  '';
```

**Comment by rycee** [comment]
Thanks! Looks good in general, I just added a few comments.

Additionally, please squash/split the commits so that there are two independent commits with the two added features. Something like

- lorri: add nixPackage option
- lorri: add support for notifying on build events

File: modules/services/lorri.nix (hunk 4 above)
Comment (nyarly) [reply modules/services/lorri.nix]: I guess the thing here is that I'm not sure how useful the script is outside the specific context of a per-user systemd service. There's some discussion in the lorri issues about whether it makes sense to standardize this as a lorri subcommand - I'm not sure there's a universal version that makes sense.

**Comment by rycee** [comment]
Thanks! I squashed and merged to master now. Note, I decided to rename the `notify` option to `enableNotifications` to make it more clear.

File: modules/services/lorri.nix (hunk 2 above)
Comment (rycee) [reply modules/services/lorri.nix]: The `nix.package` option is available now but I left changing the default as an exercise for some intrepid soul 😉

**Pull request merged**